server/src/controllers/searchController.js

```diff
@@ -0,0 +1,79 @@
+import { Op } from 'sequelize';
+import models from '../database/models';
+
+const {
+  Articles, Users, Tags
+} = models;
+/**
+ * search for an article
+ * @method customSearch
+ * @param {object} req
+ * @param {object} res
+ * @returns {(function|object)} Function next() or JSON object
+ */
+const customSearch = async (req, res) => {
```

Comment:
It might be better to break this into multiple endpoints like "/user/search?query=otse" "/articles/search?query=food"